File: C/6-08 1316(그룹 단어 체커).c
#include <stdio.h>
#include <string.h>
int main(void)
{
	char S[10000];
	int n;	
	scanf("%d", &n);
	
	int count = n;
	for (int i = 0; i < n; ++i) {
		char first = '0';
		int A[26] = {0,};
		scanf("%s", S);
		for(int j = 0; S[j] != '\0'; ++j) {
			if( first != S[j]) {
				first = S[j];
				A[S[j] - 'a'] += 1;
			}
			if (A[S[j] - 'a'] == 2) {
				count -= 1;
				break;
			}
		}
	
	}
	printf("%d\n", count);
	
	
	return 0;
}

File: python/6-08 1316(그룹 단어 체커).py
n = int(input())
count = n	

for i in range(n):
    S = input()
    for j in range(0, len(S)-1):
        if S[j] == S[j+1]:
            pass
        elif S[j] in S[j+1:]:
            count -= 1
            break
print(count)
	
